src/stdlib_str2num.fypp

```diff
@@ -109,7 +109,7 @@ module stdlib_str2num
         integer(int8) :: err
         !----------------------------------------------
         call to_num_base(s,v,p,err)
-        p = min( p , len(s, kind = int8) )
+        p = min( p , len(s) )
```

Comment:
As `p` is declared as `integer(int8)`, similar issues might still happen.  Maybe also declare `p` as `integer :: p`? Or do you think it is not needed?